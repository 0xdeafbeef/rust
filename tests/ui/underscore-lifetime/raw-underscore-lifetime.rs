// This test is to ensure that the raw underscore lifetime won't emit two duplicate errors.
// See issue #143152

//@ edition: 2021

fn f<'r#_>(){}
//~^ ERROR `_` cannot be a raw lifetime
//~| ERROR `'_` cannot be used here [E0637]

fn main() {}
