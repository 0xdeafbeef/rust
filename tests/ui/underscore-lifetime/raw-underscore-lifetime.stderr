error: `_` cannot be a raw lifetime
  --> $DIR/raw-underscore-lifetime.rs:6:6
   |
LL | fn f<'r#_>(){}
   |      ^^^^

error[E0637]: `'_` cannot be used here
  --> $DIR/raw-underscore-lifetime.rs:6:6
   |
LL | fn f<'r#_>(){}
   |      ^^^^ `'_` is a reserved lifetime name

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0637`.
